### PR TITLE
Removes lmdb and datastore modules

### PR DESCRIPTION
**Pull request by KPrasch**
**Type of PR:**
Removal

**Required reviews:** 
1

**Notes**
- Based on #2987 
- May require a configuration update script to remove the `db_filepath` field from node operator's configurations.
- Postponing a full dependency relock to isolate unrelated dependency test failures

**Why**
Resolves technical debt through removal of unused and non-essential code entities clearing the way for near-term future product development.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -11,7 +11,6 @@ python_version = "3"
 constant-sorrow = ">=0.1.0a9"
 bytestring-splitter = ">=2.4.0"
 hendrix = ">=3.4"
-lmdb = "*"
```
Comment (derekpierre) [inline Pipfile:14]: Should we relock dependencies? Or are we waiting until the code massacre is over 😄 , and then do a fresh relock?

**Review by derekpierre — APPROVED** [review]
🎸 Glad that we are finally getting rid of this.

File: Pipfile (hunk 1 above)
Comment (KPrasch) [reply Pipfile:14]: Yep - I'd like to perform a full relock in an isolated PR to avoid unrelated issues. 😅

Comment (derekpierre) [reply Pipfile:14]: 👌 Makes sense.

**Pull request merged**